diff -C 3 -r GASNet-1.26.4/other/mpi-spawner/gasnet_bootstrap_mpi.c GASNet-1.26.4-mpifix/other/mpi-spawner/gasnet_bootstrap_mpi.c
*** GASNet-1.26.4/other/mpi-spawner/gasnet_bootstrap_mpi.c	2016-07-31 23:59:23.000000000 -0700
--- GASNet-1.26.4-mpifix/other/mpi-spawner/gasnet_bootstrap_mpi.c	2016-09-21 17:39:08.958237408 -0700
***************
*** 70,75 ****
--- 70,86 ----
  void gasneti_bootstrapFini_mpi(void) {
    int err;
  
+ #if (MPI_VERSION > 1)
+   /* If somebody else has already called MPI_Finalize(), it
+    * is too late to do any MPI-related cleanup (e.g. the 
+    * MPI_Comm_free below).
+    */
+   int isfini = 0;
+   err = MPI_Finalized(&isfini);
+   gasneti_assert_always(err == MPI_SUCCESS);
+   if (isfini) return;
+ #endif
+ 
    err = MPI_Comm_free(&gasnetc_mpi_comm);
    gasneti_assert(err == MPI_SUCCESS);
  
Only in GASNet-1.26.4-mpifix/other/mpi-spawner: gasnet_bootstrap_mpi.c~
